### PR TITLE
JAVA-4421: Remove errorLabels from WriteConcernError

### PR DESCRIPTION
**Pull request by ramasai1**
[JAVA-4421](https://jira.mongodb.org/browse/JAVA-4421)

### CODE REVIEW

**Comment by stIncMale** [comment]
> Currently I am unable to run tests. I am running ./gradlew test and nothing happens.

Just in case, I am throwing everything that comes to mind regarding running the tests locally:

- JDK17.
- A accessible MongoDB listening on localhost:27017 and with enabled test commands at the very least. I create mine via [mtools](https://blog.rueckstiess.com/mtools/) with either of the three following commands:
  - `mlaunch init --single --port 27017 --storageEngine=wiredTiger --setParameter enableTestCommands=1 -v`
  - `mlaunch init --replicaset --nodes 3 --port 27017 --storageEngine=wiredTiger --setParameter enableTestCommands=1 -v`
  - `mlaunch init --sharded 2 --mongos 1 --config 1 --replicaset --nodes 3 --port 27017 --storageEngine=wiredTiger --setParameter enableTestCommands=1 -v`
- Compile and run tests:
   - `./gradlew check`

**Comment by ramasai1** [comment]
Thanks! I was running my DB on a different port and was under the impression that it was on 27017. 
Due to 2 constructors being annotated with `@Deprecated` in `MongoWriteConcernException`, there are a couple of deprecation warnings that are part of building the driver. These are in [`MongoOperationPublisher`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoOperationPublisher.java#L488) and [`MongoCollectionImpl`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-sync/src/main/com/mongodb/client/internal/MongoCollectionImpl.java#L1014). Should these be marked with `@SuppressWarnings`? Or should I be passing in an empty error labels `BsonArray`? @jyemin @stIncMale

File: driver-core/src/main/com/mongodb/bulk/WriteConcernError.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
     private final String codeName;
     private final String message;
     private final BsonDocument details;
-    private final Set<String> errorLabels;
+    @Deprecated private final Set<String> errorLabels;
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/bulk/WriteConcernError.java:38]: Moved this to the getter and add an `@deprecated` Javadoc tag.

File: driver-core/src/main/com/mongodb/internal/operation/WriteConcernHelper.java [hunk 2]
```diff
@@ -61,21 +59,18 @@ public static boolean hasWriteConcernError(final BsonDocument result) {
...
     }
 
     public static WriteConcernError createWriteConcernError(final BsonDocument writeConcernErrorDocument) {
         return new WriteConcernError(writeConcernErrorDocument.getNumber("code").intValue(),
                 writeConcernErrorDocument.getString("codeName", new BsonString("")).getValue(),
                 writeConcernErrorDocument.getString("errmsg").getValue(),
-                writeConcernErrorDocument.getDocument("errInfo", new BsonDocument()),
-                writeConcernErrorDocument.getArray("errorLabels", new BsonArray()).stream().map(i -> i.asString().getValue())
-                        .collect(Collectors.toSet()));
+                writeConcernErrorDocument.getDocument("errInfo", new BsonDocument()));
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/WriteConcernHelper.java:73]: What was the rationale for changing this?  The property is deprecated, but until it's removed it should continue to work as it does.

**Review by jyemin — CHANGES_REQUESTED** [review]
> Or should I be passing in an empty error labels BsonArray

Yes, might as well, since right after that the errorLabels are added

**Review by ramasai1 — COMMENTED** [review]
Added `@Deprecated` tags where necessary.

File: driver-core/src/main/com/mongodb/bulk/WriteConcernError.java [hunk 3]
```diff
@@ -124,8 +124,11 @@ public void addLabel(final String errorLabel) {
      *
      * @return the error labels, which may not be null but may be empty
      * @since 4.1
+     *
+     * @deprecated Prefer using error labels included in the top level response document
```
Comment (ramasai1) [inline driver-core/src/main/com/mongodb/bulk/WriteConcernError.java:128]: Copied this from the ticket because I could not think of a better note.

**Pull request merged**